src/representor/jsonapi.ts

```diff
@@ -0,0 +1,86 @@
+import Link from '../link';
+// import { resolve } from '../utils/url';
+import Representation from './base';
+
+/**
+ * A JSON:API link can either be a string, or an object with at least a href
+ * property.
+ */
+type JsonApiLink = string | { href: string };
+
+/**
+ * This type represents a valid JSON:API response. We're only interested
+ * in the links object at the moment, so everything else is (for now)
+ * untyped.
+ */
+type JsonApiObject = {
+  links?: {
+    [rel: string]: JsonApiLink | JsonApiLink[]
+  },
+  [s: string]: any
+};
+
+/**
+ * This class represents JSON:API responses.
+ *
+ * The Representor is responsible from extracting any links from the body,
+ * so they can be followed.
+ */
+export default class JsonApi extends Representation {
+
+  body: JsonApiObject;
+
+  constructor(uri: string, contentType: string, body: any) {
+
+    super(uri, contentType, body);
+
+    if (typeof body === 'string') {
+      this.body = JSON.parse(body);
+    } else {
+      this.body = body;
+    }
+
+    this.links = parseJsonApiLinks(uri, this.body);
+
+  }
+
+}
+
+/**
+ * This function takes a JSON:API object, and extracts the links property.
+ */
+function parseJsonApiLinks(baseHref: string, body: JsonApiObject): Link[] {
+
+  const result: Link[] = [];
+
+  if (body.links === undefined) {
+    return result;
+  }
+
+  for (const [rel, linkValue] of Object.entries(body.links)) {
```

Comment:
This is the obvious thing to do. But _technically_, keys of a links object do not have to be link relations. I don't know what else you could do for this until it's properly addressed upstream though, so I think it's fine.